### ISSUE TITLE
Date picker, Range picker inline mode

### ISSUE THREAD

**Issue by IonelLupu** [ant-design/ant-design#40051]
### What problem does this feature solve?

This feature will allow us to show DatePicker components (Default, Week, Month, Year, Range, Range Time, Range Week, Range, Month, Range Year) in inline mode similar to what NgZorro is doing already: https://ng.ant.design/components/date-picker/en#components-date-picker-demo-inline :
![image](https://user-images.githubusercontent.com/4083652/210846649-edbe91e5-8aca-498f-a75f-9e5677530021.png)

If will be very handy, for example, when building a responsive DatePicker.RangePicker. As we already know, the RangePicker is not responsive. A solution is to show a Modal or Drawer on mobile when you want to pick the dates. In that modal we can show the RangePicker in `inline={true}` mode and maybe style it a bit.

### What does the proposed API look like?

```
<DatePicker inline={true} />
<DatePicker.RangePicker inline={true} />
```

**Comment by zombieJ** [comment]
Current design is not well to fit for mobile device. I think we should provide a new design for this requirement instead of just provide `inline` mode.

**Comment by IonelLupu** [comment]
@zombieJ  I agree with having a new design for this but, inline mode can be also used when you want to have a date picker directly on the page without an actual input box to click. Responsiveness is just my use case.

**Comment by MadCcc** [comment]
We would extend Calendar to support your demand instead of an `inline` DatePicker. 
Maybe the boundary of DatePicker and Calendar should be explicted. cc @arvinxx

**Comment by IonelLupu** [comment]
@MadCcc as I can see from the docs, the Calendar is usually used to "display" data and the DatePicker is used for "data entry". So there is a clear distinction between the two. In my case, I need to "enter data" and that's why I am using a DatePicker.

I know AntDesign is using the [rc-picker](https://www.npmjs.com/package/rc-datepicker) package for this and they have inline examples [here](https://react-component.github.io/picker/?selectedKind=rc-picker&selectedStory=basic&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) . Maybe this can be useful 😄

**Comment by tkorkalainen** [comment]
`inline` / static / fixed DatePicker (RangePicker to be exact) would suit my needs better because I would like to use it in a custom modal directly. In addition to RangePicker component (two calendars where user could select the date range from) the modal would contain other data, too.

**Comment by satheesh3** [comment]
Do we have this feature available now ?

**Comment by bombillazo** [comment]
Hey! I too would like the rangepicker to be inlined in my component.

**Comment by kianj2** [comment]
+ 1

I don't think the Calendar component covers all the use cases that the date/range picker does. As far as I am aware the calendar can only cover day/month selection. However, I would like an inline component that can also cover weeks, quarters, years, as well as custom ranges.

**Comment by Forestych** [comment]
+1
Did anyone found a solution to use it in inline mode?

**Comment by KaziMuneeb** [comment]
Hey! I too would like the rangepicker to be inlined in my component.

**Comment by Vincent9466** [comment]
需要
